paper/paper.Rmd

```diff
@@ -0,0 +1,171 @@
+
+# Features
+
+As of this writing, `{lintr}` offers `r length(all_linters())` linters.
+
+```{r}
+library(lintr)
+
+length(all_linters())
+```
+
+Naturally, we can't discuss all of them here. To see details about all available linters, we encourage readers to see <https://lintr.r-lib.org/dev/reference/index.html#individual-linters>.
+
+We will showcase one linter for each kind of common problem found in R code.
+
+- **Best practices**
+
+`{lintr}` offers linters that can detect problematic antipatterns and suggest alternative patterns that follow best practices.
+
+For example, expressions like `ifelse(x, TRUE, FALSE)` and `ifelse(x, FALSE, TRUE)` are redundant; just `x` or `!x` suffice in R code where logical vectors are a core data structure. The `redundant_ifelse_linter()` linter detects such problematic usages.
+
+```{r}
+lint(
+  text = "ifelse(x >= 2.5, TRUE, FALSE)",
+  linters = redundant_ifelse_linter()
+)
+```
+
+```{r, eval=FALSE, echo=FALSE}
+lint(
+  text = "ifelse(x >= 2.5)",
+  linters = redundant_ifelse_linter()
+)
+```
+
+- **Efficiency**
+
+Sometimes the users might not be aware of a more efficient way offered by R for carrying out a computation. `{lintr}` offers linters to provide suggestions to improve code efficiency.
+
+```{r}
+lint(
+  text = "any(is.na(x), na.rm = TRUE)",
+  linters = any_is_na_linter()
+)
+```
+
+```{r, echo=FALSE, eval=FALSE}
+lint(
+  text = "anyNA(x)",
+  linters = any_is_na_linter()
+)
+```
+
+- **Readability**
+
+Coders spend significantly more time reading compared to writing code [@mcconnell2004code]. Thus, writing readable code makes the code more maintainable and reduces the possibility of introducing bugs stemming from a poor understanding of the code.
+
+`{lintr}` provides a number of linters that suggest more readable alternatives. For example, `function_left_parentheses_linter()`.
+
+```{r}
+lint(
+  text = "stats::sd (c (x, y, z))",
+  linters = function_left_parentheses_linter()
+)
+```
+
+```{r, echo=FALSE, eval=FALSE}
+lint(
+  text = "stats::sd(c(x, y, z))",
+  linters = function_left_parentheses_linter()
+)
+```
+
+- **Tidyverse style**
+
+`{lintr}` also provides linters to enforce the style used throughout the `{tidyverse}` [@Wickham2019] ecosystem of R packages. This style of coding has been outlined in the tidyverse style guide (https://style.tidyverse.org/index.html).
+
+```{r}
+lint(
+  text = "1:3 %>% mean %>% as.character",
+  linters = pipe_call_linter()
+)
+```
+
+```{r, echo=FALSE, eval=FALSE}
+lint(
+  text = "1:3 %>% mean() %>% as.character()",
+  linters = pipe_call_linter()
+)
+```
+
+# Benefits of using `{lintr}`
```

Comment:
One thing I'd also highlight is our design for extensibility -- yes, we enforce the Tidyverse guide by default, but we designed the package to fit many different style guides for any person/team/org that wants to deviate or even to start their own guide.

Reply:
How about this?

> `{lintr}` is designed for extensibility by allowing users to easily create custom linting rules, enabling flexible and tailored code analysis to suit specific coding conventions, style guides, and project requirements.